model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/util/SessionTimeouts.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.keycloak.models.sessions.infinispan.util;
+
+import org.keycloak.common.util.Time;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.sessions.infinispan.entities.AuthenticatedClientSessionEntity;
+import org.keycloak.models.sessions.infinispan.entities.LoginFailureEntity;
+import org.keycloak.models.sessions.infinispan.entities.UserSessionEntity;
+import org.keycloak.models.utils.SessionTimeoutHelper;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class SessionTimeouts {
+
+    /**
+     * This indicates that entry is already expired and should be removed from the cache
+     */
+    public static final long ENTRY_EXPIRED_FLAG = -2l;
+
+    /**
+     * This is used just if timeouts are not set on the realm (usually happens just during tests when realm is created manually with the model API)
+     */
+    public static final int MINIMAL_EXPIRATION_SEC = 300;
+
+    /**
+     * Get the maximum lifespan, which this userSession can remain in the infinispan cache.
+     * Returned value will be used as "lifespan" when calling put/replace operation in the infinispan cache for this entity
+     *
+     * @param realm
+     * @param userSessionEntity
+     * @return
+     */
+    public static Long getUserSessionLifespanMs(RealmModel realm, UserSessionEntity userSessionEntity) {
```

Comment:
Should this return `long` rather than `Long`? It seems never to return `null`. (similarly for the rest of this class)